### PR TITLE
Initial stab at onboarding documentation for engineering managers.

### PR DESCRIPTION
**Pull request by alloy**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+- [ ] The people-ops team should have signed you up for the various on-site manager trainings.
+- [ ] Your manager should have invited you to meetings relevant to managers.
+- [ ] Your manager should have scheduled 1:1s for you with the other engineering managers, including the CTO, and other leadership people.
+
+### Extended sprint rotation and meet & greet
+
+Unlike individual contributors, who rotate for a few sprints, engineering managers are expected to rotate across all teams. While your main responsibility will be to be a people manager, we expect anybody on the engineering team to understand roughly how we do things and what systems are in use at Artsy, but _mostly_ this will be a chance to work with all the engineering team members and product people such as PMs and designers. Currently there are 5 teams, so with sprints having a duration of 2 weeks that’s easily going to span your next 2 months.
```
Comment (sweir27) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: I think it's actually very important that the manager is familiar with our tech in order to help unblock people, facilitate discussions, and set expectations correctly. This is slightly different than the expectation for new hires. 😄 

I wonder if we can reword this to more explicitly call that out. Brainstorm idea: perhaps the goal at the end of the sprint rotation should be for the person to do a short presentation (to other managers? to everyone?) to summarize our stack and how all the different teams contribute to it.

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+During this time you will be expected to perform work as any other individual contributor, however, time will be reserved for the main objectives which are:
+
+- [ ] Have 1:1s with each and every engineer (that includes tech-leads), PM, designer, and some business owners. The tech-leads and PMs should have been made aware of this objective and provide you with time to do so. You are responsible for organizing these.
```
Comment (sweir27) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: What's the goal of the meetings with the business owners and PMs? Perhaps it could be to understand how our technology fuels a specific business. (or is it truly just to chat?)

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+- [ ] While working with individual contributors and during your 1:1s, try to uncover, and make note of, people’s expectations that are currently not being met by the organization or their manager.
+- [ ] By the end of sprint rotation you should have a list of individual contributors that you think you should manage and have argumentation for why.
+- [ ] Be involved in operational tasks which are excellent opportunities to talk to various stake-holders to gain an understanding of what needs to be communicated. For example:
```
Comment (sweir27) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: This feels like a lot-- if he is trying to get to know the stack + the individuals + the business, it seems like the first two items here can wait.

Improving the manager onboarding README does seem directly applicable (and is something that should last the entire onboarding-time). So I think we should keep that one. 😄

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    - [ ] Write a post-mortem.
+    - [ ] Work on `TODO` items that have come out of retros.
+    - [ ] Work on improving and expanding this nimble `README` repository to help shape our communication.
+- [ ] Be visible by performing tasks like:
```
Comment (sweir27) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: What's the goal of "Being visible"? I think in the beginning, he should have the opportunity to be as heads-down as he needs to be to soak in necessary information and I worry about framing this in a way that's superficial.

Maybe we can just say "Get to know our team processes and help us learn from your experiences." as a way to motivate this section?

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    - [ ] Taking turn in our [support rotation](/playbooks/support.md), possibly by keeping an eye out for people looking to trade some days.
```
Comment (sweir27) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: I think the following two checkboxes could be "required". This one is for sure optional given that people may not need to drop any shifts.

Seems to me we could drop this as it's already an action item to read over the guide/join #incidents in the other onboarding doc.

**Comment by alloy** [comment]
Great feedback, @sweir27. I’ll let it sink in a bit and iterate over the weekend.

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+During this time you will be expected to perform work as any other individual contributor, however, time will be reserved for the main objectives which are:
+
+- [ ] Have 1:1s with each and every engineer (that includes tech-leads), PM, designer, and some business owners. The tech-leads and PMs should have been made aware of this objective and provide you with time to do so. You are responsible for organizing these.
+- [ ] While working with individual contributors and during your 1:1s, try to uncover, and make note of, people’s expectations that are currently not being met by the organization or their manager.
+- [ ] By the end of sprint rotation you should have a list of individual contributors that you think you should manage and have argumentation for why.
```
Comment (alloy) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: @dblock Remind me, what was the motivation for this? I assumed it was meant as an exercise for him to think about how he can add value to a report as their manager, is that the case? If so, I think we can maybe drop this one and improve [the above](https://github.com/artsy/README/pull/95/files#diff-f85005588f9a5181ef9ec4d27af69920R22) to “…, try to develop an understanding of where and how you want to add value as a manager”

Comment (alloy) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: While I don’t believe the intent was to use this list as if set in stone, but just for the exercise, it may give off the wrong signal.

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md (hunk 1 above)
Comment (alloy) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: I like that objective 👍  I _think_ it would be good to present it to everyone, both to contribute to presence _and_ so there’s more people that can provide feedback about potential misunderstandings?

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md (hunk 2 above)
Comment (alloy) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: Both. The write-up was way to implicit 👍

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md (hunk 4 above)
Comment (alloy) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: Besides being good to learn about our processes, these types of tasks are usually the ones that help bring visibility to a wider audience and an external manager can do with some support in that area to more quickly solidify their name and ability into people’s minds. Having said that, I think that framing it the way you’ve suggested still implicitly achieves that without being done in a superficial way.

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+The goal of onboarding is to set you up for success as a new engineering manager at Artsy. To that end, your main objective during this time will be to gather knowledge. You will be given ample time to get acquainted with your new peers, our business, and our tech.
+
+_For now we’ll try to target roughly two months, as there are currently 5 product teams and sprints have a duration of 2 weeks. Whether or not this duration was too arbitrary is something we’ll have to learn._
+
+To help work on the main objective, we’ve tried to identify the following goals.
+
+### Get to know Artsy's team
+
+- [ ] Have 1:1s with each and every engineer (that includes tech-leads) to get to know the people, their interests, their north-star goals, and their perspectives on the businesses they contribute to.
```
Comment (ashkan18) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: as a possible suggestions, would it be easier if these 1:1s where setup for each team members of the current rotation? Like if you are in Sell this Sprint, setup 1:1s with all sell team members during this sprint and etc.

Comment (ashkan18) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: we could also suggest few famous Artsy 1:1s places if Adam is not familiar with the neighborhood :)

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+- [ ] By the end of your sprint rotation, put together a short presentation that summarizes our stack and how different teams contribute to it. (Target audience TBD, although I _think_ it would be good to present it to everyone, both to contribute to presence _and_ so there’s more people that can provide feedback about potential misunderstandings.)
+
+### Get to know Artsy's business
+
+- [ ] Have 1:1s with business owners to make your acquaintance and gain an understanding of what the landscape in their piece of the business looks like in general and how our technology uniquely fuels our take on it.
+
+- [ ] Have 1:1s with PMs to gain an understanding of how product and engineering work together to produce results in their respective parts of the business.
+
+- [ ] TODO: Flesh out: go to a gallery and auction. Besides being fun, what could they learn from doing this and in what way should they do so? E.g. shadow a liaison, casually visit?
```
Comment (alloy) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: (In my opinion, this TODO doesn’t need to block a merge, we can figure that out together as we go.)

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md (hunk 7 above)
Comment (alloy) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: I think the other person in each of his 1:1s will probably show him around the neighborhood. But if there are specific advantages for choosing some places over others for one’s 1:1, then that would be helpful.

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+## After your first week
+
+You’ve done general people-ops and engineering team onboarding, and have casually met some team-members. Now it’s time to start onboarding as a new engineering manager.
+
+- [ ] If you’re reading this, you’re the first external engineering manager we’ve hired and your name is Adam. Yay! (If not, this will be _real_ awkward.)
+
+     At Artsy we are trying to give new hires ownership over improving the onboarding documentation that they are faced with, in an effort to help the next new hire onboard more smoothly. As you are the first one, there is plenty of work to do in this area; in fact, this very document is nothing but a first draft. Things in this document may turn out to be insufficient, too broad, or simply not work at all. That’s all good and you should feel empowered to bring this up openly.
+
+     It will be your task to make this document into a substantially larger one, that draws from all the things you had to learn. You’re not on your own, your fellow peers will all be available to support you during this task. You got this!
```
Comment (joeyAghion) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: Maybe doesn’t have to be “substantially larger” but could definitely be “improved.”

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+### Get to know Artsy's tech
+
+- [ ] Unlike individual contributors, who rotate teams for a few sprints to gain a general understanding of how we work as contributors, engineering managers are expected to rotate across _all_ the teams to gain the understanding needed to successfully support their reports in achieving their goals and growth. (For example, unblocking people by knowing how to achieve something or who/where to delegate to, facilitate discussions by understanding context and who to involve for a given topic; and how to set expectations for engineers, product, and business owners as to what is realistically achievable.) Tech-leads and PMs should have been made aware of your main objective and provide you with time to do so.
+
+- [ ] By the end of your sprint rotation, put together a short presentation that summarizes our stack and how different teams contribute to it. (Target audience TBD, although I _think_ it would be good to present it to everyone, both to contribute to presence _and_ so there’s more people that can provide feedback about potential misunderstandings.)
```
Comment (joeyAghion) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: This deliverable would rightly intimidate anyone, given the complexity and variety of the platform. Maybe that’s OK, but I’d consider ways of shrinking its scope.

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+## After your first week
+
+You’ve done general people-ops and engineering team onboarding, and have casually met some team-members. Now it’s time to start onboarding as a new engineering manager.
+
+- [ ] If you’re reading this, you’re the first external engineering manager we’ve hired and your name is Adam. Yay! (If not, this will be _real_ awkward.)
+
+     At Artsy we are trying to give new hires ownership over improving the onboarding documentation that they are faced with, in an effort to help the next new hire onboard more smoothly. As you are the first one, there is plenty of work to do in this area; in fact, this very document is nothing but a first draft. Things in this document may turn out to be insufficient, too broad, or simply not work at all. That’s all good and you should feel empowered to bring this up openly.
+
+     It will be your task to improve this document, drawing from all the things you’ve had to learn. You’re not on your own, your fellow peers will all be available to support you during this task. You got this!
```
Comment (ashfurrow) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: `this document` should be a link to https://github.com/artsy/README/blob/master/.github/ISSUE_TEMPLATE/engineering-manager-onboarding.md

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
+---
+name: Engineering Manager Onboarding
+about: A list of things to do as a new manager to become familiar with the nature of our engineering team and the context it exists in.
+---
+
+## After your first week
+
+You’ve done general people-ops and engineering team onboarding, and have casually met some team-members. Now it’s time to start onboarding as a new engineering manager.
```
Comment (starsirius) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: Should we make this more like a template?

Comment (starsirius) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]:
```suggestion
You’ve done general people-ops and engineering team onboarding, and have casually met some team members. Now it’s time to start onboarding as a new engineering manager.
```

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+- [ ] If you’re reading this, you’re the first external engineering manager we’ve hired and your name is Adam. Yay! (If not, this will be _real_ awkward.)
```
Comment (starsirius) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: Should we make this more like a template?

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md (hunk 12 above)
Comment (starsirius) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]:
```suggestion
You’ve done general people-ops and engineering team onboarding, and have casually met some team members. Now it’s time to start onboarding as a new engineering manager.
```

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+- [ ] While working with individual contributors and during your 1:1s, try to develop an understanding of where and how you want to add value as a manager.
+
+- [ ] Keep a continuous and open discussion going with your peer managers about opportunities for people you think you could manage and why. By the end of sprint rotation we should organically have arrived at a set of suggestions that we, as the entire engineering team, can _consider_.
+
+### Get to know Artsy's tech
+
+- [ ] Unlike individual contributors, who rotate teams for a few sprints to gain a general understanding of how we work as contributors, engineering managers are expected to rotate across _all_ the teams to gain the understanding needed to successfully support their reports in achieving their goals and growth. (For example, unblocking people by knowing how to achieve something or who/where to delegate to, facilitate discussions by understanding context and who to involve for a given topic; and how to set expectations for engineers, product, and business owners as to what is realistically achievable.) Tech-leads and PMs should have been made aware of your main objective and provide you with time to do so.
```
Comment (starsirius) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: Maybe we can move "engineering managers are expected to rotate across _all_ the teams" expectation to the top of this "Your first ~two months" section, given it's main format to know Artsy's team, tech, and business.

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md (hunk 10 above)
Comment (starsirius) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: Do we keep Artsy technology stack somewhere besides the [blog post](http://artsy.github.io/blog/2017/04/14/artsy-technology-stack-2017/)? I think it'll will be useful to use this opportunity to keep updating it, e.g. the diagram (maybe still with teams in it) below, and encourage discussion with the whole team.

screen shot 2018-10-30 at 10 32 18 am

**Comment by starsirius** [comment]
Woohoo! Very exciting! Thanks for putting this together! 🎉

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+- [ ] The people-ops team should have signed you up for the various on-site manager trainings.
+
+- [ ] Your manager should have invited you to meetings relevant to managers.
+
+- [ ] Your manager should have scheduled 1:1s for you with the other engineering managers, including the CTO, and other leadership people.
+
+## Your first ~two months
+
+The goal of onboarding is to set you up for success as a new engineering manager at Artsy. To that end, your main objective during this time will be to gather knowledge. You will be given ample time to get acquainted with your new peers, our business, and our tech.
```
Comment (dblock) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: I think it's both knowledge and, very importantly, _context_.

I would add _strategy_ here as well, as the managers are the messengers of it, often.

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+To help work on the main objective, we’ve tried to identify the following goals.
+
+### Get to know Artsy's team
+
+- [ ] Have 1:1s with each and every engineer (that includes tech-leads) to get to know the people, their interests, their north-star goals, and their perspectives on the businesses they contribute to.
+
+- [ ] While working with individual contributors and during your 1:1s, try to develop an understanding of where and how you want to add value as a manager.
+
+- [ ] Keep a continuous and open discussion going with your peer managers about opportunities for people you think you could manage and why. By the end of sprint rotation we should organically have arrived at a set of suggestions that we, as the entire engineering team, can _consider_.
+
```
Comment (dblock) [inline .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: The title of "Get to know Artsy's team" implies beyond Engineering, but we talk about 1:1s outside of engineering later. So maybe this should be the Engineering team?

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md (hunk 10 above)
Comment (orta) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: @dleve123 is interested in building something like ^ using a normal architecture diagramming tools (as opposed to something like the above that requires pretty deep sketch knowledge )

Comment (sweir27) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: I think original goal of this was just to have some outcome/deliverable from the mega-rotation. If it's not this specific thing, it could be TBD (or nothing at all).

Comment (dleve123) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: Re https://github.com/artsy/README/pull/95/files#r229330904: @joeyAghion and I are brainstorming an architecture diagram project for the Platform Practice

Generally, I'm pro having some concrete deliverable/session for someone new. IDK if it should be on the topic of the sprint rotation though (I do think that's pretty intimating), it could just be on a topic that the new manager wants to present upon (maybe something they see as a gap in Artsy).

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md (hunk 14 above)
Comment (alloy) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: I hear what you’re saying and also had it like that in an earlier iteration, but seeing as we’re calling out opportunities to meet with the team and business people separately, I think it’s ok to scope this to the tech aspect. If you or others feel more strongly about this then let’s revisit.

**Comment by alloy** [comment]
@dblock Excellent points.

> Hiring. This is a top priority for a manager, therefore they should get involved with hiring in some capacity during onboarding. This may include sitting on a reference call or actually interviewing a candidate.

What do you think of making the item be to be a hiring manager and add it to the “help the team get to know you” section, as they’ll have to work with a diverse set of people? As part of this they can sit in on a reference call and actually do an interview.

> Direct reports and team. Managers are taking direct reports at some point. At which point? How do we determine that? I think opening into that or including this as part of onboarding, even as a placeholder for the first hire to figure out is important.

True. I only alluded to the definition of their team as “we’ve jointly put together your team”, but can make that it’s own item. As discussed previously with @sweir27, putting together the team should be an ongoing and open discussion and so I do think that it should be mostly a placeholder to be fleshed out from our future experience, but maybe it does make sense to call out that _after_ sprint rotation we’ll have final discussions? Just to make sure we don’t accidentally try to do that at the end of sprint rotation when they should still be focussed on gaining knowledge from the product team they’re on by that time.

> Managers are the beholders of Artsy values. Understanding what that means in Engineering is important, not sure how to express this.

In the general onboarding checklist, that the manager works through in their first week, we already call out both Artsy’s values _and_ [the engineering principles](https://github.com/artsy/README/blob/master/culture/engineering-principles.md). Stressing on it here, to me, kinda feels like it’s of less importance to ICs, which I would disagree with. On the other hand, accidentally reading past some items [in one’s first week] is very understandable, so maybe this one does  bear repeating? I’m on the fence.

File: .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md (hunk 15 above)
Comment (alloy) [reply .github/ISSUE_TEMPLATE/engineering-manager-onboarding.md]: Took a stab at this, let me know if it came across as you intended.

**Comment by dblock** [comment]
> @dblock Excellent points.
> 
> > Hiring. This is a top priority for a manager, therefore they should get involved with hiring in some capacity during onboarding. This may include sitting on a reference call or actually interviewing a candidate.
> 
> What do you think of making the item be to be a hiring manager and add it to the “help the team get to know you” section, as they’ll have to work with a diverse set of people? As part of this they can sit in on a reference call and actually do an interview.

Yes, and/but I think we should require that a new manager sits on a ref call or an interview as a fly on the wall and generally familiarize themselves with engineering hiring. Whichever way you want to phrase this.

> > Direct reports and team. Managers are taking direct reports at some point. At which point? How do we determine that? I think opening into that or including this as part of onboarding, even as a placeholder for the first hire to figure out is important.
> 
> True. I only alluded to the definition of their team as “we’ve jointly put together your team”, but can make that it’s own item. As discussed previously with @sweir27, putting together the team should be an ongoing and open discussion and so I do think that it should be mostly a placeholder to be fleshed out from our future experience, but maybe it does make sense to call out that _after_ sprint rotation we’ll have final discussions? Just to make sure we don’t accidentally try to do that at the end of sprint rotation when they should still be focussed on gaining knowledge from the product team they’re on by that time.

Makes sense.

> > Managers are the beholders of Artsy values. Understanding what that means in Engineering is important, not sure how to express this.
> 
> In the general onboarding checklist, that the manager works through in their first week, we already call out both Artsy’s values _and_ [the engineering principles](https://github.com/artsy/README/blob/master/culture/engineering-principles.md). Stressing on it here, to me, kinda feels like it’s of less importance to ICs, which I would disagree with. On the other hand, accidentally reading past some items [in one’s first week] is very understandable, so maybe this one does bear repeating? I’m on the fence.

No strong feelings.
